R/c3.R

```diff
@@ -147,29 +148,66 @@ c3 <- function(data,
     #this option is currently only for grouped scatter plots
 
     # remove columns not in x,y,group
-    data <- data[, c(x, y, group)]
+    data2 <- data[, c(x, y, group)]
 
-    groups <- as.character(unique(data[[group]]))
+    groups <- as.character(unique(data2[[group]]))
 
-    tmp.df <- group_by_at(data, group) %>%
+    tmp.df <- group_by_at(data2, group) %>%
       mutate(id = 1:n())
 
     # need to change columns to group, group_x in xs and in data dataframe
-    data <- data.table::dcast(setDT(tmp.df),
+    data2 <- data.table::dcast(setDT(tmp.df),
                               formula(sprintf('id ~ %s', group)),
                               value.var = c(y,x))
-    data <- as.data.frame(data[, -1]) # need to remove id column (always in pos 1)
+    data2 <- as.data.frame(data2[, -1]) # need to remove id column (always in pos 1)
```

Comment:
Hi, I think your idea and approach will work but the tests are currently failing. Instead of naming data2 just replace the data object. Make sure you run the package tests before submitting

Reply:
I haven't run package tests before, happy to do so but can you point out some instructions for this?

Reply:
this package uses `testthat` and the test are alreaqdy written. To run them just use `devtools::test()` or in rstudio use the build tab >> more >> test package